src/main/ruby/core/signal.rb

```diff
@@ -43,14 +43,13 @@ module Signal
     Numbers[number] = name
   end
 
-  # Define CLD as CHLD if it's not defined by the platform
-  unless Names.key? 'CLD'
-    Names['CLD'] = Names['CHLD']
-  end
+  # Define aliases if it's not defined by the platform
+  Names['CLD'] = Names['CHLD'] unless Names.key? 'CLD'
+  Names['IOT'] = Names['ABRT'] unless Names.key? 'IOT'
```

Comment:
It seems SIGIOT is never missing on our supported platforms though.
Does MRI have such handling?

Reply:
I've seen the failure on my machine. It was random and not reproducible. Since I knew how it was better to fix.

Reply:
> Does MRI have such handling?

Reply:
I don't know, you are free to have a look. How is it relevant to this?

Reply:
Well, where does this remapping come from?
If it's some bug due to some randomly failed build I'd rather not commit that.

Reply:
I wrote it based on information on a wiki page where IOT was mentioned as alias of ABRT.
Not sure what you mean by the second sentence.

Reply:
This just seems impossible (all our supported platforms define both IOT and ABRT).
I don't want to add workarounds for unknown build bugs.
Also, this has nothing to do with this PR, please remove this commit from this PR.
It could be related to SVM not accepting all signals, but these two seem to work fine in SVM.

Reply:
I've got it during running a rails application on JVM few times, no svm involved. As agreed it will stay in this PR. Asking again, what do you mean by the broken build? Where did you look up that it's supported? I was getting Unknown signal from sun.misc.Signal.

Reply:
[This line](https://github.com/graalvm/truffleruby/blob/ed917326a1a8f62a628afd9d9b95055f85910019/src/main/ruby/core/signal.rb#L40) iterates every native config starting with `platform.signal.` and all of Linux, macOS and Solaris define `platform.signal.SIGIOT` in their native configuration, so `"IOT"` is __*always*__ in `Names`.

So I would bet a lot this doesn't actually fix the problem observed.
Yet you are pushing to get this commit in with no proof it solves anything.
Even if it fixes something, it's definitely not the right fix.
`Unknown signal from sun.misc.Signal` has nothing to do with the `Names` hash, it's a different issue.

#982 is probably closer to the fix (it fixes the case where `trap(signal, "DEFAULT")` would be called, no handler was ever set and `signal` is not recognized by `sun.misc.Signal`).
The proper fix has to be in Java code, or rescue the error around the primitive `vm_watch_signal` call.

> As agreed it will stay in this PR.

As agreed where?
I'm not asking to split in another, I'm asking to remove this commit which is not solving anything (and also totally unrelated).
I will make sure this PR will not be merged with this commit in.